app/views/PillarAnalysis/AnalyticalStatementInput/StoryAnalysisModal/index.tsx

```diff
@@ -315,6 +341,106 @@ function StoryAnalysisModal(props: Props) {
         message: 'Looks like there are some changes that have not been saved yet. Are you sure you want to close?',
     });
 
+    const variables = useMemo(() => (
+        isDefined(automaticSummaryId)
+            ? {
+                projectId,
+                summaryId: automaticSummaryId,
+            }
+            : undefined
+    ), [projectId, automaticSummaryId]);
+
+    const {
+        data,
+        loading,
+        startPolling,
+        stopPolling,
+        error,
+    } = useQuery<AnalyticalInformationSummaryQuery, AnalyticalInformationSummaryQueryVariables>(
+        ANALYTICAL_INFORMATION_SUMMARY,
+        {
+            skip: !automaticSummaryId,
+            variables,
+        },
+    );
+
+    const informationGapOnClick = useCallback(() => {
+        if (pristine) {
+            setPristine(false);
+        }
```

Comment:
This check is not necessary. Regardless of what the pristine is, you can set it to new value. Resetting to new value for boolean, interger and string won't rerender the component.